controllers/user/utils.js

```diff
@@ -0,0 +1,73 @@
+const _ = require('lodash');
+const { body } = require('express-validator/check');
+
+const userBasePath = '/user';
+
+const userEndpoints = {
+    dashboard: '/dashboard',
+    register: '/register',
+    login: '/login',
+    logout: '/logout',
+    activate: '/activate',
+    requestpasswordreset: '/request-password-reset',
+    resetpassword: '/reset-password'
+};
+
+const makeUserLink = page => userBasePath + userEndpoints[page];
+
+// convert a single error string into a list
+// or return an express-validator pre-formatted list
+const makeErrorList = error => {
+    if (_.isArray(error)) {
+        return error;
+    } else {
+        return [
+            {
+                msg: error
+            }
+        ];
+    }
+};
+
+// generic function to return the user to the form with an error
+const renderUserError = (msg, req, res, path) => {
+    if (msg) {
+        req.flash('formErrors', makeErrorList(msg));
+    }
+    req.session.save(() => {
```

Comment:
Why is this saving the session here? Particularly in a utils function I'm wary of something so stateful.